website/static/js/fangorn.js

```diff
@@ -1345,6 +1345,113 @@ function doCheckout(item, checkout, showError) {
     });
 }
 
+/**
+ * Submit to Boa in Action Column
+ * @param event DOM event object for click
+ * @param {Object} item A Treebeard _item object for the row involved. Node information is inside item.data
+ * @param {Object} col Information pertinent to that column where this upload event is run from
+ * @private
+ */
+function _submitToBoaEvent (event, item, col) {
+    var tb = this;
+    var selectedDataset = '2019 October/GitHub';
+    function updateDataset(dataset) {
+        selectedDataset = dataset;
+    }
+    function cancelSubmitToBoa() {
+        tb.modal.dismiss();
+    }
+    function runSubmitToBoa(item, dataset) {
+        return $osf.postJSON(
+            item.data.nodeApiUrl + 'boa/submit-job/',
+            {
+                data: item.data,
+                parent: item.parent().data,
+                dataset: dataset,
+            }
+        ).done(function(xhr) {
+            $osf.growl(
+                'Success',
+                'File submitted to Boa. You will be notified by email when the job is done.',
+                'success'
+            );
+            tb.modal.dismiss();
+            // if (showError) {
+            //     window.location.reload();
+            // }
+        }).fail(function(xhr) {
+            console.error('@@@   BOA FAIL!', item);
+            $osf.growl(
+                'Error',
+                'Unable to submit the file to Boa. Please try again later. ' +
+                'Contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.'
+            );
+            tb.modal.dismiss();
+        });
+    }
+
+    var datasets = [
```

Comment:
How feasible would it be to make this DRY, wrt to the other hard-coded list in `boa/settings/defaults.py`?

Reply:
Out-of-scope, similarly, a future work if requested by them (e.g. they need us to update the dataset).